core/src/main/java/org/apache/iceberg/Partitioning.java

```diff
@@ -211,6 +211,7 @@ public static StructType partitionType(Table table) {
 
     Map<Integer, PartitionField> fieldMap = Maps.newHashMap();
     List<NestedField> structFields = Lists.newArrayList();
+    Map<Integer, Integer> fieldToIndex = Maps.newHashMap();
```

Comment:
I still think this needs a new name ... since we only use it to track indexes of void transformed fields

Reply:
Updated the name, and left comments here.